### PR TITLE
chore: implement internal version of hcl.Traversal

### LINKED ISSUE #42: Fix HCL fork
Users would need to use https://github.com/jlarfors/hcl

That would complicated things

### PR DESCRIPTION
**Pull request by jlarfors**
Removing the dependence on hcl.Traversal removes the need to use a
forked and patched version of hcl to support the hcl.TraverseSplat

resolves #42

### CODE REVIEW

File: .copywrite.hcl [hunk 1]
```diff
@@ -12,6 +12,8 @@ project {
     ".vscode/**",
     "bin/**",
     "**/.lingon/**",
+    "**/.terra/**",
+    "**/.terriyaki/**",
```
Comment (veggiemonk) [inline .copywrite.hcl:16]: ?

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 # Local .lingon directories (generated code)
 **/.lingon/*
 **/.terriyaki/*
```
Comment (veggiemonk) [inline .gitignore:6]: ?

File: example/go.mod [hunk 3]
```diff
@@ -4,8 +4,6 @@ go 1.20
 
 replace github.com/volvo-cars/lingon => ../
 
-replace github.com/hashicorp/hcl/v2 => github.com/jlarfors/hcl/v2 v2.16.0-patch-1
-
```
Comment (veggiemonk) [inline example/go.mod:8]: 👏

File: pkg/terra/reference.go [hunk 4]
```diff
@@ -77,76 +79,220 @@ var _ tkihcl.Tokenizer = (*Reference)(nil)
...
+type referenceStepType int
+
+const (
+	referenceStepAttribute = 1
+	referenceStepIndex     = 2
+	referenceStepKey       = 3
+	referenceStepSplat     = 4
+)
+
+type referenceStep struct {
```
Comment (veggiemonk) [inline pkg/terra/reference.go:207]: I feel we can start moving those important struct at the top of the file.
I was confused the first time I read the file.

File: .copywrite.hcl (hunk 1 above)
Comment (jlarfors) [reply .copywrite.hcl]: We had an example (maybe platypus?) generating `.terra` directories. I removed `.terriyaki` now as that should no longer exist.

File: .gitignore (hunk 2 above)
Comment (jlarfors) [reply .gitignore]: Same as above

> We had an example (maybe platypus?) generating `.terra` directories. I removed `.terriyaki` now as that should no longer exist.

File: pkg/terra/reference.go (hunk 4 above)
Comment (jlarfors) [reply pkg/terra/reference.go]: Done. Moved to under the `Reference` struct declaration. Good idea.

File: .copywrite.hcl (hunk 1 above)
Comment (veggiemonk) [reply .copywrite.hcl]: yeah I was questioning the .terriyaki stuff, sorry I should have been more clear

**Review by veggiemonk — APPROVED** [review]
well done!

**Pull request merged**